### PR TITLE
Feature/anoncreds

### PR DESCRIPTION
**Pull request by Artemkaaas**

### CODE REVIEW

File: libindy/src/commands/anoncreds/issuer.rs [hunk 1]
```diff
@@ -109,23 +100,26 @@ impl IssuerCommandExecutor {
                                           schema_json: &str,
                                           signature_type: Option<&str>,
                                           create_non_revoc: bool) -> Result<String, IndyError> {
-        let schema = Schema::from_json(schema_json)
-            .map_err(map_err_trace!())
+        info!(target: "commands/anoncreds/issuer", "create_and_store_claim_definition >>> wallet_handle: {:?}, issuer_did: {:?}, schema_json: {:?}, \
```
Comment (jovfer) [inline libindy/src/commands/anoncreds/issuer.rs]: should be trace without specific target

File: libindy/tests/anoncreds.rs [hunk 2]
```diff
@@ -1430,321 +1390,321 @@ mod demos {
...
-        let valid = AnoncredsUtils::verifier_verify_proof(&proof_req_json,
-                                                          &proof_json,
-                                                          &schemas_json,
-                                                          &claim_defs_json,
-                                                          &revoc_regs_jsons).unwrap();
-        assert!(valid);
-
-        TestUtils::cleanup_storage();
-    }
+    //    #[cfg(feature = "interoperability_tests")]
```
Comment (jovfer) [inline libindy/tests/anoncreds.rs]: Uncomment and disable feature in pipeline

File: wrappers/java/indy.iml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jovfer) [inline wrappers/java/indy.iml]: remove this file from PR

File: Jenkinsfile.cd [hunk 4]
```diff
@@ -257,7 +258,7 @@ def windowsTesting() {
 def ubuntuTesting() {
     node('ubuntu') {
         stage('Ubuntu Test') {
-            linuxTesting("ci/ubuntu.dockerfile ci", "Ubuntu", true, "pool_network")
+            linuxTesting("ci/ubuntu.dockerfile ci", "Ubuntu", false, "pool_network")
```
Comment (ashcherbakov) [inline Jenkinsfile.cd]: Why is it false now?

File: libindy/src/api/anoncreds.rs [hunk 5]
```diff
@@ -494,7 +494,7 @@ pub extern fn indy_prover_store_claim(command_handle: i32,
 /// #Returns
 /// claims json
 ///     [{
-///         "claim_uuid": <string>,
+///         "referent": <string>,
 ///         "attrs": [{"attr_name" : "attr_value"}],
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Please specify what values do we return: encoded or not or both

File: libindy/src/api/anoncreds.rs [hunk 6]
```diff
@@ -538,26 +538,26 @@ pub extern fn indy_prover_get_claims(command_handle: i32,
 ///         "name": string,
 ///         "version": string,
 ///         "nonce": string,
-///         "requested_attr1_uuid": <attr_info>,
-///         "requested_attr2_uuid": <attr_info>,
-///         "requested_attr3_uuid": <attr_info>,
-///         "requested_predicate_1_uuid": <predicate_info>,
-///         "requested_predicate_2_uuid": <predicate_info>,
+///         "requested_attr1_referent": <attr_info>,
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Please specify how <attr_info> looks like

File: libindy/src/api/anoncreds.rs [hunk 7]
```diff
@@ -538,26 +538,26 @@ pub extern fn indy_prover_get_claims(command_handle: i32,
...
+///         "requested_attr1_referent": [claim1, claim2],
+///         "requested_attr2_referent": [],
+///         "requested_attr3_referent": [claim3],
+///         "requested_predicate_1_referent": [claim1, claim3],
+///         "requested_predicate_2_referent": [claim2],
 ///     }, where claim is
 ///     {
-///         "claim_uuid": <string>,
+///         "referent": <string>,
 ///         "attrs": [{"attr_name" : "attr_value"}],
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Please specify what values do we return: encoded or not or both

File: libindy/src/api/anoncreds.rs [hunk 8]
```diff
@@ -607,43 +607,58 @@ pub extern fn indy_prover_get_claims_for_proof_req(command_handle: i32,
 /// proof_req_json: proof request json as come from the verifier
 ///     {
 ///         "nonce": string,
-///         "requested_attr1_uuid": <attr_info>,
-///         "requested_attr2_uuid": <attr_info>,
-///         "requested_attr3_uuid": <attr_info>,
-///         "requested_predicate_1_uuid": <predicate_info>,
-///         "requested_predicate_2_uuid": <predicate_info>,
+///         "requested_attr1_referent": <attr_info>,
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Please specify what values do we return: encoded or not or both

File: libindy/src/api/anoncreds.rs [hunk 9]
```diff
@@ -607,43 +607,58 @@ pub extern fn indy_prover_get_claims_for_proof_req(command_handle: i32,
...
+///         "claim1_referent_in_wallet": <revoc_reg1>,
+///         "claim2_referent_in_wallet": <revoc_reg2>,
+///         "claim3_referent_in_wallet": <revoc_reg3>,
 ///     }
 /// cb: Callback that takes command result as parameter.
 ///
+/// where attr_info:
+///     {
+///         "name": attribute name,
+///         "schemas_seq_no": a list of schema sequence numbers allowed for proving of the attribute
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Probably it's better to specify a list of issuer DIDs for **each** schema_seq_no?

File: libindy/src/api/anoncreds.rs [hunk 10]
```diff
@@ -607,43 +607,58 @@ pub extern fn indy_prover_get_claims_for_proof_req(command_handle: i32,
...
+///         "issuer_dids":  a list of issuer dids allowed for proving of the attribute
+///     }
+/// predicate_info:
+///     {
+///         "attr_name": attribute name,
+///         "p_type": predicate type (Currently GE only)
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Did we rename GE to something else?

File: libindy/src/services/anoncreds/helpers.rs [hunk 11]
```diff
@@ -1,242 +1,45 @@
...
-        let mut nums = vec![
-            BigNumber::from_hex("ff9d2eedfee9cffd9ef6dbffedff3fcbef4caecb9bffe79bfa94d3fdf6abfbff").unwrap().to_bytes().unwrap(),
-            BigNumber::from_hex("ff9d2eedfee9cffd9ef6dbffedff3fcbef4caecb9bffe79bfa9168615ccbc546").unwrap().to_bytes().unwrap()
-        ];
-        let res = get_hash_as_int(&mut nums);
-
-        assert!(res.is_ok());
-        assert_eq!("9E2A0653691B96A9B55B3D1133F9FEE2F2C37B848DBADF2F70DFFFE9E47C5A5D", res.unwrap().to_hex().unwrap());
+    for predicate in predicates_for_claim {
+        sub_proof_request_builder.add_predicate(&predicate.attr_name, "GE", predicate.value)?;
```
Comment (ashcherbakov) [inline libindy/src/services/anoncreds/helpers.rs]: We need to rename GE to something more human readable?

**Pull request merged**